core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.ts

```diff
@@ -346,6 +346,20 @@ angular.module('oppia').component('learnerDashboardPage', {
         );
       };
 
+      /**
+       * This was needed beacause javascript returns -1 using inBuilt indexOf
+       * even when there are identical objects.
+       * Refer -> https://rb.gy/kh5ry5
+       */
+      var findIndex = (arr, element) => {
+        for (var i = 0; i < arr.length; i++) {
+          if (JSON.stringify(arr[i]) === JSON.stringify(element)) {
+            return i;
+          }
+        }
+        return -1;
+      };
```

Comment:
Also want to highlight this. Is there a better approach to this?

/cc @kevintab95

Reply:
Typo: because

Hm, this doesn't look great. Do we really need to compare the entire object in this case? If not, can we use something like this:

```ctrl.incompleteExplorationsList.findIndex(exp => exp.id === activity.id)```

Reply:
But nice find! Wonder why this didn't result in a bug so far...

Reply:
> Do we really need to compare the entire object in this case?

I'm not sure too.

Reply:
Also, in general, always add the complete URL in the codebase. SO answers/questions generally have short URLs.

Reply:
Actually the complete url was longer than 80 chars :sweat_smile:

Reply:
Oh click on the share button and copy the link. They are already shortened.
eg. https://stackoverflow.com/q/11704971, https://stackoverflow.com/a/11705017

Reply:
Please take a look at the code and figure out if entire object needs to be compared. If not, use the method described above. Thanks!

Reply:
done